compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -467,6 +467,17 @@ private void checkXMLNamespacePrefixes(List<BLangXMLElementFilter> filters) {
         }
     }
 
+    private int getPreferredMemberTypeTag(BFiniteType finiteType) {
+        for (BLangExpression valueExpr : finiteType.getValueSpace()) {
+            for (int i = TypeTags.INT; i <= TypeTags.DECIMAL; i++) {
+                if (valueExpr.getBType().tag == i) {
+                    return i;
+                }
+            }
+        }
+        return 0;
```

Comment:
Probably clearer to return something like `org.wso2.ballerinalang.compiler.util.TypeTags#NONE`.